actors/builtin/paych/paych_test.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestPaymentChannelActor_Constructor(t *testing.T) {
+	ctx := context.Background()
+	actor := pcActorHarness{Actor{}, t}
+
+	pcaAddr := tutil.NewIDAddr(t, 100)
+	callerAddr := tutil.NewIDAddr(t, 101)
+
+	t.Run("can create a payment channel actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
+			WithReceiverType(builtin.AccountActorCodeID)
+
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt, pcaAddr, callerAddr)
+	})
+
+	t.Run("fails if caller is not account actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).WithCaller(callerAddr, builtin.CronActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID)
+		rt.ExpectAbort(18, func() {
+			rt.Call(actor.Constructor, &ConstructorParams{To: pcaAddr})
+		})
+	})
+
+	t.Run("fails if target is not account actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
+			WithReceiverType(builtin.CronActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID)
+		expMsg := "target actor [0 - 64] must be an account (bafkqadlgnfwc6mjpmfrwg33vnz2a), was bafkqactgnfwc6mjpmnzg63q"
+		rt.ExpectAbortWithMsg(16, expMsg, func() {
+			rt.Call(actor.Constructor, &ConstructorParams{To: pcaAddr})
+		})
+	})
+
+	t.Run("fails if addr is not ID type", func(t *testing.T) {
+		pcaAddr1 := tutil.NewActorAddr(t, "beach blanket babylon")
+		builder := mock.NewBuilder(ctx, pcaAddr1).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
+			WithReceiverType(builtin.AccountActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID)
+		expMsg := "target address must be an ID-address, [2 - 343095ece71e03006965c10e88314fb7c103f413] is 2"
+		rt.ExpectAbortWithMsg(16, expMsg, func() {
+			rt.Call(actor.Constructor, &ConstructorParams{To: pcaAddr1})
+		})
+	})
+}
+
+func TestPaymentChannelActor_UpdateChannelState(t *testing.T) {
+	ctx := context.Background()
+	actor := pcActorHarness{Actor{}, t}
+
+	pcaAddr := tutil.NewIDAddr(t, 100)
+	callerAddr := tutil.NewIDAddr(t, 101)
+	syscalls := tutil.MockSyscalls{VerifiesSig: true}
+
+	t.Run("Can add a lane/update actor state", func(t *testing.T) {
+		balance:= abi.NewTokenAmount(100)
+		received := abi.NewTokenAmount(0)
+		builder := mock.NewBuilder(ctx, pcaAddr).
+			WithBalance(balance, received).
+			WithEpoch(abi.ChainEpoch(2)).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
+			WithReceiverType(builtin.AccountActorCodeID).
+			WithSysCalls(&syscalls)
+
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt, pcaAddr, callerAddr)
+		rt.ExpectValidateCallerAddr(callerAddr, pcaAddr)
```

Comment:
Nit: move the expectations as close to the matching `Call` as possible. This location makes it seem related to the previous construction call.